src/functions-and-arrays.js

```diff
@@ -24,18 +44,45 @@ function sum() {}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(arrays) {
+if (!arrays.length) return null;
+
+let sum = 0;
+for (let i = 0; i < arrays.length; i++) {
+  sum += arrays[i];
+
+}
+return sum / arrays.length; 
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(Arrays) { 
+if (!Arrays.length) return null;
+
+let word = 0;
+for ( let i = 0; i < Arrays.length; i++) {
+  //if (word.length > Arrays[i].length) {
+    word += Arrays[i].length; 
+  //} 
+}
+return word / Arrays.length; 
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(arr) {
+  if(!arr.length) return null
+
+  else {
+    return sum(arr) / arr.length
```

Comment:
Good use of a previous function, saving lines of code instead of repeating. Just found it odd because you didn't complete the iteration 3.1 maybe you deleted it by mistake.